verilog/analysis/verilog_linter.cc

```diff
@@ -206,28 +210,53 @@ std::vector<LintRuleStatus> VerilogLinter::ReportStatus(
   return statuses;
 }
 
-LinterConfiguration LinterConfigurationFromFlags() {
-  LinterConfiguration config;
-
-  // Turn on default ruleset.
-  const auto& ruleset = absl::GetFlag(FLAGS_ruleset);
-  config.UseRuleSet(ruleset);
+const std::string linter_config_file = ".rules.verible_lint";
 
+bool AppendLinterConfigurationFromFile(
+    LinterConfiguration &config, absl::string_view config_filename) {
   // Read local configuration file
   std::string content;
-  if (verible::file::GetContents(absl::GetFlag(FLAGS_rules_config), &content)) {
+  if (verible::file::GetContents(config_filename, &content)) {
     RuleBundle local_rules_bundle;
     std::string error;
     if (local_rules_bundle.ParseConfiguration(content, &error)) {
       config.UseRuleBundle(local_rules_bundle);
     } else {
-      LOG(ERROR) << "Unable to fully parse configuration: " << error
-                 << std::endl;
+      LOG(WARNING) << "Unable to fully parse configuration: " << error;
+    }
+    return true;
+  }
+
+  return false;
+}
+
+LinterConfiguration LinterConfigurationFromFlags(absl::string_view filename) {
+  LinterConfiguration config;
+
+  // Turn on default ruleset.
+  const auto& ruleset = absl::GetFlag(FLAGS_ruleset);
```

Comment:
Since this is new code, can we avoid accessing global flags here, and pass their values in?  It makes testing that much easier.

Reply:
It's not completely new - the `LinterConfigurationFromFlags` already existed.

I've added a TODO to clean up this function in the PR with waiver files. I will be happy to clean this up once one of these PRs is merged (I'd rather not do this earlier as both my PRs modify this function, but neither introduces it, so to avoid conflicts I'd rather at least one of them was merged before cleaning this up).

Reply:
OK.